fuel/datasets/base.py

```diff
@@ -28,10 +29,6 @@ class Dataset(object):
         'targets')`` for MNIST (regardless of which data the data stream
         actually requests). Any implementation of a dataset should set this
         attribute on the class (or at least before calling ``super``).
-    default_iteration_scheme : :class:`.IterationScheme`, optional
```

Comment:
Hold on: `DataStream(dataset)` works only for the `IterableDataset` currently, right? This is exactly the problem that I encountered: how can I simply iterate through the examples of a `IndexableDataset`?

I would argue that there is no such thing as the reasonable size, and `DataStream(dataset)` should always be the stream of examples. For this reason I would keep the default scheme feature but use the `ExampleScheme` as the default for the `IndexableDataset`.